src/library/scala/collection/immutable/NumericRange.scala

```diff
@@ -132,6 +132,59 @@ sealed class NumericRange[T](
   // are forgiving: therefore the checks are with the methods.
   private def locationAfterN(n: Int): T = start + (step * fromInt(n))
 
+  private def crossesTheEndAfterN(n: Int): Boolean = {
+    // if we're sure that subtraction in the context of T won't overflow, we use this function
+    // to calculate the length of the range
+    def unsafeRangeLength(r: NumericRange[T]): T = {
+      val diff = num.minus(r.end, r.start)
+      val quotient = num.quot(diff, r.step)
+      val remainder = num.rem(diff, r.step)
+      if (!isInclusive && num.equiv(remainder, num.zero)) quotient else num.plus(quotient, num.one)
+    }
+
+    if (num.equiv(start, end) || n == 0) return n >= 1
+
+    val smallerThanInt = num match {
+      case _: Numeric.ShortIsIntegral | _: Numeric.ByteIsIntegral => true
```

Comment:
Couldn't you just do the whole calculation here, exactly, with `Int`s, and be done with that part rather than coming back to it later?

Reply:
I actually wrote it this way at first, then I realized having a general algorithm/solution would be more reliable and readable rather than writing different logics for different generic types. We just leave the final calculations type-specific, and the logic stays the same for every type `T`. Also, that way I had to implement two different `unsafeLength` methods, one which works for small types and returns `Int`, another one which returns `T`. Let me know if you think the other way.

Reply:
The unity of having a single algorithm sounds like a benefit.  Makes sense to keep it this way, then!

Reply:
Alright, resolving this as well!